### PR TITLE
Add metrics

### PR DESCRIPTION
**Pull request by bwwinthehouse**

### CODE REVIEW

File: lib/kafka_rest_client.js [hunk 1]
```diff
@@ -151,6 +154,9 @@ KafkaRestClient.prototype.produce = function produce(produceMessage, callback) {
         });
     } else {
         self.produceWithRetry(produceMessage, 0, callback);
+        if (this.statsd) {
+          this.statsd.increment(self.metricsPrefix + '.produced'+ produceMessage.topic);
```
Comment (abansal) [inline lib/kafka_rest_client.js]: In the naming convention I used for python event (produced/retry/latency)is in the end

File: lib/kafka_rest_client.js [hunk 2]
```diff
@@ -182,6 +188,10 @@ KafkaRestClient.prototype.produceWithRetry = function produceWithRetry(produceMe
                 'TimeStamp': produceMessage.timeStamp.toString()
             };
             httpClient.post(reqOpts, produceMessage.message, function handlePostCall(err, res, body) {
+                if (this.statsd && typeof this.statsd.timing === 'function') {
```
Comment (fx19880617) [inline lib/kafka_rest_client.js]: change this to self should pass the test

File: lib/kafka_rest_client.js [hunk 3]
```diff
@@ -181,7 +187,12 @@ KafkaRestClient.prototype.produceWithRetry = function produceWithRetry(produceMe
                 'Content-Type': supportContentType[produceMessage.type],
                 'TimeStamp': produceMessage.timeStamp.toString()
             };
-            httpClient.post(reqOpts, produceMessage.message, function handlePostCall(err, res, body) {
+            httpClient.post(reqOpts, produceMessage.message, function handlePostCall(err, res, body, duration) {
+                var metricPrefix = self.metricsPrefix + produceMessage.topic;
+                if (self.statsd && typeof self.statsd.timing === 'function') {
+                  self.statsd.timing(metricPrefix + '.latency', duration);
```
Comment (fx19880617) [inline lib/kafka_rest_client.js]: Also put topic name inside the metric structure

Comment (bwwinthehouse) [inline lib/kafka_rest_client.js]: ```var metricPrefix = self.metricsPrefix + produceMessage.topic;```

It has it

File: lib/kafka_rest_client.js [hunk 4]
```diff
@@ -181,7 +187,13 @@ KafkaRestClient.prototype.produceWithRetry = function produceWithRetry(produceMe
                 'Content-Type': supportContentType[produceMessage.type],
                 'TimeStamp': produceMessage.timeStamp.toString()
             };
+            var time = new Date();
             httpClient.post(reqOpts, produceMessage.message, function handlePostCall(err, res, body) {
+                var metricPrefix = self.metricsPrefix + produceMessage.topic;
+                if (self.statsd && typeof self.statsd.timing === 'function') {
+                  self.statsd.timing(metricPrefix + '.latency', time);
```
Comment (fx19880617) [inline lib/kafka_rest_client.js]: Is there a way to measure the whole e2e client latency?

**Comment by fx19880617** [comment]
Overall lgtm

**Pull request merged**